sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTableScanExec.scala

```diff
@@ -535,4 +535,23 @@ class GpuHiveDelimitedTextPartitionReader(conf: Configuration,
       }
     }
   }
+
+  /**
+   * Override of [[com.nvidia.spark.rapids.GpuTextBasedPartitionReader.castStringToDate()]],
+   * to convert parsed string columns to Dates.
+   * Two key differences from the base implementation, to comply with Hive LazySimpleSerDe
+   * semantics:
+   *   1. The input strings are not trimmed of whitespace.
+   *   2. Invalid date strings do not cause exceptions.
+   */
+  override def castStringToDate(input: ColumnVector, dt: DType): ColumnVector = {
+    val cudfFormat = DateUtils.toStrf(dateFormat, parseString = true)
```

Comment:
Do we need to validate the date format works before we put this on the GPU? Is it even configurable in hive?  Should we just hard code it instead of relying on the CSV options to pick the format?

Reply:
It did not seem configurable. I'd better double-check, given that I got it wrong for `Timestamp`.

Reply:
Or I got it wrong it timestamp. There are multiple times that I have read the code and what actually happens is different.  Like with the date parsing, so I don't know if I can trust myself.